scripts/globals/fishing_contest.lua

```diff
@@ -0,0 +1,417 @@
+    [ 7] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 8] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 9] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [10] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [11] = {              item = xi.item.PELICAN_RING,                               },
+    [12] = {              item = xi.item.PELICAN_RING,                               },
+    [13] = {              item = xi.item.PELICAN_RING,                               },
+    [14] = {              item = xi.item.PELICAN_RING,                               },
+    [15] = {              item = xi.item.PELICAN_RING,                               },
+    [16] = {              item = xi.item.PELICAN_RING,                               },
+    [17] = {              item = xi.item.PELICAN_RING,                               },
+    [18] = {              item = xi.item.PELICAN_RING,                               },
+    [19] = {              item = xi.item.PELICAN_RING,                               },
+    [20] = {              item = xi.item.PELICAN_RING,                               },
+}
+
+xi.fishingcontest.interval =
+{
+    -- These intervals indicate how long each phase lasts before being automatically progressed.
+    -- Total cycle is 28 days, 35 minutes based on captures.  May be varied if contest is manually controlled
+    [xi.fishingcontest.status.CONTESTING] = 300,        -- 00 days 00 hours 05 mins
+    [xi.fishingcontest.status.OPENING]    = 1500,       -- 00 days 00 hours 25 mins
+    [xi.fishingcontest.status.ACCEPTING]  = 1209600,    -- 14 days 00 hours 00 mins
+    [xi.fishingcontest.status.RELEASING]  = 1800,       -- 00 days 00 hours 30 mins
+    [xi.fishingcontest.status.PRESENTING] = 1206000,    -- 13 days 23 hours 00 minutes
+    [xi.fishingcontest.status.HIATUS]     = 2100,       -- 00 days 00 hours 35 minutes
+    [xi.fishingcontest.status.CLOSED]     = 0xFFFFFFFF, -- Must be manually controlled
+}
+
+-----------------------------------
+-- LOCAL FUNCTIONS
+-----------------------------------
+
+-- Determine the slot that contains the fish being traded
+local function findFishSlot(trade, fish)
+    for i = 0, trade:getSlotCount() - 1 do
+        if trade:getItemId(i) == fish then
+            return i
+        end
+    end
+
+    return 0
+end
+
+-- The npcUtil.giveItem function currently does not support custom exdata
+-- Since we need to include the measurements of the fish, we need a modified function
+local function giveFish(player, params)
+    params = params or {}
+    local ID = zones[player:getZoneID()]
+    local fishid = params.id
+    -- does player have enough inventory space?
+    if player:getFreeSlotsCount() < 1 then
+        local messageId = (ID.text.ITEM_CANNOT_BE_OBTAINED + 4) or ID.text.ITEM_CANNOT_BE_OBTAINED
```

Comment:
I don't understand what's happening here? In Lua the only things that are falsy are nil and false, not 0. So if any part of ID.text.ITEM_CANNOT_BE_OBTAINED is nil, then so will the other or side

Reply:
I copied this from the npcUtils function to be consistent as an interim (as mentioned before, the npcUtil functions don't currently handle exdata so I had to make my own for now):
![image](https://github.com/user-attachments/assets/89d192a1-158e-4787-bb49-9a06b08021dc)

I can adjust this, but I'm not really sure what to adjust it to if the goal is to guarantee that the ITEM_CANNOT_BE_OBTAINED is not null (or more specifically, equals a particular value)